OPSEXP-2242: enable inbound smtp service (#123)
Alfresco/alfresco-helm-charts: charts/alfresco-repository/docs/email.md

# Configuring Email related features

> Alfresco repository can interact with users through emails i different ways.

Reviewer: ```suggestion
Alfresco repository can interact with users through emails in different ways.
```

Below we describe how to enable and configure the 3 main email features Alfresco
offers:

* Sending emails
* Accepting incoming email
* IMAP interface to the repository

## Sending emails

> For Alfresco to send email it needs to talk to an mail relay administrators
> must configure. This mail relay is considered part of the required architecture
> and admin can configure it using regular Alfresco properties.

Reviewer: ```suggestion
To enable Alfresco to send emails, it needs to communicate with a mail relay
that administrators must configure. This mail relay is considered an integral
part of the required architecture, and administrators can configure it using
standard Alfresco properties.
```

There are 2 ways to pass properties to the repository:

* Using `configuration.repository.existingConfigMap`
* Using `environment.CATALINA_OPTS` env var.

### Using a configmap

Passing properties through a configmap is as simple as setting it in the
alfresco-global.properties file. Start by creating a file called
`alfresco-global.properties` as a regular properties file:

```properties
mail.host=mysmtp.infra.local
mail.port=587
mail.username=alfresco
mail.from.default=alfresco@domain.tld
mail.smtp.auth=true
mail.smtp.starttls.enable=true
```

> Note: Although possible, we do not set the authentication password is not set
> in the configmap. Instead we will create a secret for it.

We then create a configmap from the file with the mail properties (and possibly
other properties):

```bash
kubectl create configmap repo-config \
  --from-file=alfresco-global.properties
```

### Using a env vars

> Passing properties through the CATALINA_OPTS environment variable is done as
> shown below:

Reviewer: ```suggestion
Passing properties through the CATALINA_OPTS environment variable can be done
as shown below:
```

```properties
environment:
  CATALINA_OPTS: >-
    -Dmail.host=mysmtp.infra.local
    -Dmail.port=587
    -Dmail.username=alfresco
    -Dmail.from.default=alfresco@domain.tld
    -Dmail.smtp.auth=true
    -Dmail.smtp.starttls.enable=true
```

### Passing credentials

> Regardless of whether you chose to pass properties using a configmap or
> directly values, you should not pass credentials that same way.

Reviewer: ```suggestion
Regardless of whether you choose to pass properties using a configmap or
directly values, you should avoid passing credentials in the same way.
```

To securely pass credentials we will create a secret so the Alfresco repository
can authenticate to the mail relay:

```bash
kubectl create secret generic mail-secret \
  --from-literal "MAIL_PASSWORe=mysupersecret"
```

> Now when calling this charts use the properties below:

Reviewer: ```suggestion
Now when installing the chart you can use the values below:
```

        service.beta.kubernetes.io/load-balancer-source-ranges: 10.0.0.0/24
        service.beta.kubernetes.io/aws-load-balancer-nlb-target-type: instance
        service.beta.kubernetes.io/aws-load-balancer-backend-protocol: tcp
        service.beta.kubernetes.io/aws-load-balancer-subnets: subnet-xxxx, subnet-yyyy
        service.beta.kubernetes.io/aws-load-balancer-ssl-cert=arn:aws:acm:region:account:certificate/certificate_ID
environment:
  CATALINA_OPTS: >-
    -Demail.inbound.unknownUser=anonymous
    -Demail.server.domain=myecm.domain.tld
    -Demail.server.allowed.senders=ecm-ingestor@myecm.domain.tld
    -Demail.server.blocked.senders=.*
    -Demail.server.auth.enabled=true
```

Here we configure an SMTPs service to listen on port 1465 through an AWS
loadbalancer and accept emails from a single sender. This loadbalancer will be
responsible for doing the SSL off-load and further traffic will be cleartext.
This is suitable in most cases as - as said earlier - your SMTP server should
never be exposed directly on internet and have an SMTP proxy in front of it.

In case you really need to terminate the SSL session on the Alfresco repo see
the [keystore doc](./keystores.md)

> * IMAP interface to the repository

Reviewer: ```suggestion
### Enabling IMAP
```
